File: backend/src/main/java/com/mylibrary/entity/Emprestimo.java
package com.mylibrary.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "emprestimos")
@Getter
@Setter
public class Emprestimo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "livro_id", nullable = false)
    private Livro livro;

    @Column(nullable = false)
    private String nomePessoa;

    private String telefone;

    @Column(nullable = false)
    private LocalDate dataEmprestimo;

    @Column(nullable = false)
    private LocalDate dataDevolucaoPrevista;

    private LocalDate dataDevolucaoEfetiva;
}

File: frontend/src/app/models/emprestimo.ts
import { Livro } from './livro';

export interface Emprestimo {
  id?: number;
  livro: Livro;
  nomePessoa: string;
  telefone: string;
  dataEmprestimo?: string;
  dataDevolucaoPrevista: string;
  dataDevolucaoEfetiva?: string;
}
